src/solidity/TransferableAccountStore.sol

```diff
@@ -98,75 +104,44 @@ contract TransferableAccountStore is Suapp, ITransferableAccountStore {
      * @param accountId The account ID
      * @param _address The address to approve
      */
-    function approveAddressCallback(
-        SignatureVerifier.TimedSignature calldata timedSignature,
-        Suave.DataId accountId,
-        address _address
-    ) public emitOffchainLogs onlyLocked(Utils.iToHex(abi.encodePacked(accountId))) {
-        require(_verifyTimedSignature(timedSignature), "Invalid timedSignature");
-        require(
-            isOwner(Utils.iToHex(abi.encodePacked(accountId)), timedSignature.signer),
-            "The signer is not the owner of the account."
-        );
-        accountApprovals[accountId] = _address;
-        emit AddressApproved(Utils.iToHex(abi.encodePacked(accountId)), _address);
-    }
-
-    /**
-     * @dev Approve an address for a given account
-     * @param accountId The account ID
-     * @param _address The address to approve
-     * @return bytes The encoded callback data
-     */
     function approveAddress(
         SignatureVerifier.TimedSignature calldata timedSignature,
         string memory accountId,
         address _address
-    ) external view onlyLocked(accountId) returns (bytes memory) {
-        require(_verifyTimedSignature(timedSignature), "Invalid timedSignature");
-
+    ) external onlyLocked(accountId) {
+        if (!verifyTimedSignature(timedSignature)) {
+            revert InvalidTimedSignature();
+        }
         Account storage account = accountsStore[accountId];
-        require(account.owner == timedSignature.signer, "Only owner can approve addresses");
-        return abi.encodePacked(
-            this.approveAddressCallback.selector, abi.encode(timedSignature, account.accountId, _address)
-        );
+        if (!isOwner(accountId, timedSignature.signer)) {
+            revert OnlyOwnerCanApproveAddresses();
+        }
```

Comment:
In the future, we could improve code readability by creating custom modifiers for common checks like `onlyVerified` and `onlyOwner`. This would reduce code duplication and make function bodies cleaner. Not a high priority, but worth considering for future refactoring.

examples:

```solidity
error InvalidTimedSignature();
error OnlyOwnerCanPerform();

modifier onlyVerified(SignatureVerifier.TimedSignature calldata timedSignature) {
    if (!verifyTimedSignature(timedSignature)) {
        revert InvalidTimedSignature();
    }
    _;
}

modifier onlyOwner(string memory accountId, address signer) {
    if (!isOwner(accountId, signer)) {
        revert OnlyOwnerCanPerform();
    }
    _;
}
```